src/librustdoc/clean/mod.rs

```diff
@@ -1763,10 +1763,9 @@ impl Clean<GenericArgs> for hir::GenericArgs<'_> {
     fn clean(&self, cx: &mut DocContext<'_>) -> GenericArgs {
         if self.parenthesized {
             let output = self.bindings[0].ty().clean(cx);
-            GenericArgs::Parenthesized {
-                inputs: self.inputs().clean(cx),
-                output: if output != Type::Tuple(Vec::new()) { Some(output) } else { None },
-            }
+            let output =
+                if output != Type::Tuple(Vec::new()) { Some(Box::new(output)) } else { None };
```

Comment:
Personally I would use `box output` here :shrug: but it won't affect performance in practice, `output` is already on the stack.

Oh, this is the bit that breaks https://github.com/rust-lang/rust/issues/83718#issuecomment-811453368 ! I've been looking for it.

Reply:
> Personally I would use `box output` here 🤷

Yeah, I thought about that, but someone recently removed all the uses of `box_syntax` from rustdoc and other tools; I think the goal is to get rid of `box_syntax` eventually. IIRC from the discussion there, the newest version of LLVM has an optimization that gets rid of the perf difference, but I may be misremembering.

Reply:
> Oh, this is the bit that breaks [#83718 (comment)](https://github.com/rust-lang/rust/issues/83718#issuecomment-811453368) ! I've been looking for it.

Hooray! 🎉